src/generate.c

```diff
@@ -292,6 +310,27 @@ int main(int argc, char** argv)
         FILE* f = fopen(generator_run_stamp, "w");
         g_assert(f != NULL);
         fclose(f);
+    } else if (check_called_just_in_time()) {
+        /* When booting with cloud-init, network configuration
+         * might be provided just-in-time. Specifically after
+         * system-generators were executed, but before
+         * network.target is started. In such case, auxiliary
+         * units that netplan enables have not been included in
+         * the initial boot transaction. Detect such scenario and
+         * add all netplan units to the initial boot transaction.
+         */
+        if (any_networkd) {
+            start_unit_jit("systemd-networkd.socket");
+            start_unit_jit("systemd-networkd-wait-online.service");
+            start_unit_jit("systemd-networkd.service");
```

Comment:
What about NetworkManager?

Reply:
All service units created by netplan (OVS/WPA) are only related to the networkd backend, so no special handling is needed for NetworkManager.

Reply:
In my mind, NetworkManager should be enabled/started in a similar fashion too. Because for example, we may have on Ubuntu Core cases where cloud-init provides netplan yaml, that expects to start ModemManager and NetworkManager.

And I still ideally want to have i.e. both pre-installed in the image (or cloud-init-local installing them) and enabling things on the fly. Because after this is done, I want to have the ability to install packages during boot too specifically to support OVS if it's not seeded in the image.

Reply:
But I think NetworkManager may need to come separately, later. As i.e. we don't currently generate enable for NM, nor return something like "any_networkmanager" flag to see if we need to start it.